content/shmem_symmetric_available.tex

```diff
@@ -0,0 +1,33 @@
+\apisummary{
+Returns the memory available on the symmetric heap in bytes.}
+
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void shmem_symmetric_available(size_t *available_size);
+\end{Csynopsis}
+
+\begin{apiarguments}
+\apiargument{OUT}{available\_size}{The symmetric heap memory available in bytes.}
+\end{apiarguments}
+
+
+\apidescription{
+The \FUNC{shmem\_symmetric\_available} call returns the memory available on the symmetric heap in bytes.
+The available memory is the unallocated memory on the symmetric heap, which could be allocated by using memory allocation routines such as \FUNC{shmem\_malloc} or \FUNC{shmem\_calloc}. 
```

Comment:
I'm not sure a simple 'memory available in bytes' is helpful.  For example the existing dmapp symmetric heap stats api/structure:

The dmapp_sheap_stats_t structure contains symmetric heap usage information and is returned by dmapp_sheap_stats.

typedef struct {
uint64_t size;           
uint64_t usage;    
uint64_t hwm;  
uint64_t biggest_used;  
uint64_t biggest_free;  
uint64_t used_blks;  
uint64_t free_blks; 
uint64_t allocs;  
uint64_t frees;  
uint64_t fails; 
uint64_t reserved[8]; 
  } dmapp_sheap_stats_t;

5.2.3.2 Members
size |   | Total heap size in bytes.
usage |   | Current usage in bytes.
hwm |   | High water mark in bytes
biggest_used |   | Size of largest allocator block in use
biggest_free |   | Size of largest free allocator block.
used_blks |   | Number of allocator blocks in use.
free_blks |   | Number of free allocator blocks.
allocs |   | Running total number of alloc calls.
frees |   | Running total number of free calls.
fails |   | Running total number of failed alloc calls.
reserved[8] |   | Reserved for future use.